pwiz_tools/Skyline/Controls/Graphs/GraphSpectrum.cs

```diff
@@ -598,14 +610,72 @@ public bool ShouldShowMirrorPlot
 
         public IEnumerable<SpectrumDisplayInfo> AvailableSpectra
         {
-            get { return _spectra; }
+            get { return _spectra?.SelectMany(s => s.Spectra); }
+        }
+
+        public class PrecursorSpectra
+        {
+            public PrecursorSpectra(TreeNodeMS selectedTreeNode, Peptide peptide, Target lookupTarget,
+                ExplicitMods lookupMods, TransitionGroupDocNode precursor)
+            {
+                SelectedTreeNode = selectedTreeNode;
+                Peptide = peptide;
+                LookupTarget = lookupTarget;
+                LookupMods = lookupMods;
+                Precursor = precursor;
+                Spectra = new List<SpectrumDisplayInfo>();
+                _prositCacheSettings = null;
+                _prositSpectra = new Dictionary<Tuple<IsotopeLabelType, int>, SpectrumDisplayInfo>();
+            }
+
+            private TreeNodeMS SelectedTreeNode { get; }
```

Comment:
It looks like SelectedTreeNode is only getting used for checking whether the selected item is a PeptideGroupTreeNode.
It probably isn't a good idea to hang onto a TreeNodeMS, since that's a control which belongs to the SequenceTree. I would recommend just having an IdentityPath here instead.